auto_backup/models/db_backup.py

```diff
@@ -161,7 +161,12 @@ def action_backup(self):
                     # Generate new backup
                     else:
                         with rec.custom_tempdir():
-                            db.dump_db(self.env.cr.dbname, destiny)
+                            try:
+                                db.dump_db(self.env.cr.dbname, destiny)
+                            except:  # noqa
+                                # Avoid filling up disk with failed backups
+                                os.remove(destiny)
```

Comment:
I'm musing if half of a dump couldn't be a live saver if that's the only backup there is. What's your take on this?

Reply:
I see that side of the story, but I also see the following case:

1. User has configured 3 full backups to be held at all times
2. Healthy backups are taken Monday, Tuesday and Wednesday
3. Thursday, Friday, Saturday and Sunday the backup is interrupted, therefore incomplete
4. On Monday the issue is seen and resolved, the backup runs fully, and the older backups are deleted. Saturday, Sunday and Monday are kept, but only Monday is complete
5. Tuesday people want to restore a file from before Monday morning Lucy the secretary deleted some invoices, but they can't, because they only have corrupted backups from before Monday

But I also see your point.

A sensible approach here could maybe be that we rename incomplete backups to "_incomplete". We don't count them in the retention of backups, and we keep only the most recent one.

What do you think?

Reply:
hmm, that's also non-obvious behavior. Given we send a mail for failures, users should be aware that things are wrong.

You'll need to rebase now #1623 is merged.